examples/Cargo.toml

```diff
@@ -6,13 +6,18 @@ workspace = ".."
 edition = "2018"
 
 [features]
-default = []
+default = ["debug-assertions"]
 metal = ["gfx-backend-metal"]
-gl = ["gfx-backend-gl", "gfx-backend-gl/glutin"]
-wgl = ["gfx-backend-gl", "gfx-backend-gl/wgl"]
+gl = ["gfx-backend-gl", "gfx-backend-gl/surfman"] # GL for Unix
+x11 = ["gfx-backend-gl/x11"]
+webgl = ["gfx-backend-gl"] # GL for web
+wgl = ["gfx-backend-gl", "gfx-backend-gl/wgl"] # GL for Windows
 dx11 = ["gfx-backend-dx11"]
 dx12 = ["gfx-backend-dx12"]
 vulkan = ["gfx-backend-vulkan"]
+# this is necessary because cargo doesn't allow using `target.'cfg(debug-assertions)'.dependencies`
+# to conditionally include crates
+debug-assertions = ["env_logger"]
```

Comment:
Is there already a built-in `debug-assertions` feature?

I wasn't exactly sure how to handle that, but the way it was previously setup with `target.'cfg(debug-assertions)'.dependencies` to gate the dependency was triggering a warning from Cargo that said that using the `target.cfg` feature like that wasn't supported for `debug-assertions`.

Reply:
yes, in the code `cfg(debug_assertion)` is enabled in debug builds. Now that you are declaring it here, it will probably not conflict (i.e. it will be `cfg(feature = "debug_assertions")`, but I also don't understand how we'd be enabling this now.

Reply:
Yeah, the problem is that you can't include dependencies based on the `cfg(debug-assertions)`. The line in the cargo toml was previously attempting to only include `env_logger` when `cfg(debug-assertions)` was true, but that isn't supported in Cargo, so the only way to actually parameterize the inclusion of `env_logger` was to make a feature for it. Otherwise we could just always include `env_logger` and not make it optional. It is just an example and in a real-life situation you likely won't only be logging in debug mode anyway.

Reply:
let's just depend on env_logger unconditionally in the examples, no big deal

Reply:
Sounds good.